### PR TITLE
Optional evmSigner, PolkadotService asset fix

### PR DESCRIPTION
**Pull request by nohaapav**
### Description

Please explain the changes this PR addresses here.

### Checklist

- [ ] If this requires a documentation change, I have created a PR in [moonbeam-docs](https://github.com/PureStake/moonbeam-docs) repository.
- [x] If this requires it, I have updated the Readme
- [x] If necessary, I have updated the examples
- [x] I have verified if I need to create/update unit tests
- [x] I have verified if I need to create/update acceptance tests
- [x] If necessary, I have run acceptance tests on this branch in CI

### CODE REVIEW

**Comment by nohaapav** [comment]
@ekenigs Hi there :) This PR enables user to use SDK without specifying evmSigner in case no evm based parachain is used.
I've slightly changed PolkadotService asset getter as it was not reflecting config service, yet reading static data. I believe it should check chain related assets, not entire collection.

File: packages/sdk/src/polkadot/PolkadotService.ts [hunk 1]
```diff
@@ -65,7 +64,7 @@ export class PolkadotService {
       throw new Error('No native symbol key found');
     }
 
-    const asset = assetsMap.get(key);
+    const asset = this.chain.assetsData.get(key)?.asset;
```
Comment (ekenigs) [inline packages/sdk/src/polkadot/PolkadotService.ts:67]: To remove the `assetsMap` dependency is correct but the change itself is not. As for `assetsData` in a chain, I meant to use it only if I need to overwrite something default for an asset. So it means that a chain may have not all assets in `assetsData` that a user can send. We need to find another solution.

**Review by ekenigs — CHANGES_REQUESTED** [review]
@nohaapav Hey, thanks again for your effort. `EvmSigner` part looks good but for `assetsData` we need to find another solution.

File: packages/sdk/src/polkadot/PolkadotService.ts (hunk 1 above)
Comment (nohaapav) [reply packages/sdk/src/polkadot/PolkadotService.ts:67]: @ekenigs I see, alright. We can still pass ConfigService to PolkadotService.. We already do pass all opts from SDK down to getTransferData. WDYT ?

Comment (ekenigs) [reply packages/sdk/src/polkadot/PolkadotService.ts:67]: Yes, I think it's a good idea.

Comment (nohaapav) [reply packages/sdk/src/polkadot/PolkadotService.ts:67]: i'll push the fix this week :) was bit busy

Comment (nohaapav) [reply packages/sdk/src/polkadot/PolkadotService.ts]: @ekenigs Pushed the fix. Sorry for delay, was busy with other work, finally got to xcm.

Comment (ekenigs) [reply packages/sdk/src/polkadot/PolkadotService.ts]: @nohaapav no problem, I can understand that, I'm also very busy with other projects right now. Your changes look good. Please add the changeset and I will approve.

**Comment by ekenigs** [comment]
@nohaapav Hey, can you please add `npm run changeset` so we can merge this PR, thanks.

**Comment by nohaapav** [comment]
> @nohaapav Hey, can you please add `npm run changeset` so we can merge this PR, thanks.

Done.

**Review by ekenigs — APPROVED** [review]

**Pull request merged**